src/manage_sql.c

```diff
@@ -43755,6 +43755,109 @@ openvas_default_scanner_host ()
                      SCANNER_UUID_DEFAULT);
 }
 
+/**
+ * @brief Create a new connection to an OSP scanner relay.
+ *
+ * @param[in]   host     Original host name or IP address.
+ * @param[in]   port     Original port.
+ * @param[in]   ca_pub   Original CA certificate.
+ *
+ * @return New connection if success, NULL otherwise.
+ */
+static osp_connection_t *
+osp_scanner_relay_connect (const char *host, int port, const char *ca_pub)
+{
+  int ret, new_port;
+  gchar *new_host, *new_ca_pub;
+  osp_connection_t *connection;
+
+  new_host = NULL;
+  new_ca_pub = NULL;
+  new_port = 0;
+
+  ret = slave_get_relay (host,
+                         port,
+                         ca_pub,
+                         "OSP",
+                         &new_host,
+                         &new_port,
+                         &new_ca_pub);
+
+  switch (ret)
+    {
+      case 0:
+        break;
+      case 1:
+        g_warning ("No relay found for Scanner at %s:%d", host, port);
+        return NULL;
+      default:
+        g_warning ("%s: Error getting relay for Scanner at %s:%d",
+                   __FUNCTION__, host, port);
+        return NULL;
+    }
+
+  connection
+    = osp_connection_new (new_host, new_port, new_ca_pub, NULL, NULL);
+
+  if (connection == NULL)
+    {
+      if (new_port)
+        g_warning ("Could not connect to relay at %s:%d"
+                    " for Scanner at %s:%d",
+                    new_host, new_port, host, port);
+      else
+        g_warning ("Could not connect to relay at %s"
+                    " for Scanner at %s:%d",
+                    new_host, host, port);
+    }
+
+  g_free (new_host);
+  g_free (new_ca_pub);
+
+  return connection;
+}
+
+/**
+ * @brief Create a new connection to an OSP scanner using the scanner data.
+ *
+ * @param[in]   host     Host name or IP address.
+ * @param[in]   port     Port.
+ * @param[in]   ca_pub   CA certificate.
+ * @param[in]   key_pub  Public key.
+ * @param[in]   key_priv Private key.
+ *
+ * @return New connection if success, NULL otherwise.
+ */
+osp_connection_t *
+osp_connect_with_data (const char *host,
+                       int port,
+                       const char *ca_pub,
+                       const char *key_pub,
+                       const char *key_priv)
+{
+  osp_connection_t *connection;
+  int is_unix_socket = (host && *host == '/') ? 1 : 0;
+
+  if (is_unix_socket == 0
+      && get_relay_mapper_path ())
+    {
+      connection = osp_scanner_relay_connect (host, port, ca_pub);
+    }
+  else
+    {
+      connection = osp_connection_new (host, port, ca_pub, key_pub, key_priv);
+
+      if (connection == NULL)
+        {
+          if (is_unix_socket)
```

Comment:
These if cases should be the other way round, right?

Reply:
Yes, the statements for the `if (is_unix_socket)` condition and the `else` should now be the other way with a268ef5ad18f58ef0c8d2a9f3d214c9a818f77f3.